core/app/models/spree/unit_cancel.rb

```diff
@@ -34,20 +34,6 @@ def adjust!
   # This method is used by Adjustment#update to recalculate the cost.
   def compute_amount(line_item)
     raise "Adjustable does not match line item" unless line_item == inventory_unit.line_item
-
-    -weighted_line_item_amount(line_item)
-  end
-
-  private
-
-  def weighted_line_item_amount(line_item)
-    quantity_of_line_item = quantity_of_line_item(line_item)
-    raise ZeroDivisionError, "Line Item does not have any inventory units available to cancel" if quantity_of_line_item.zero?
-
-    line_item.total_before_tax / quantity_of_line_item
-  end
-
-  def quantity_of_line_item(line_item)
-    BigDecimal(line_item.inventory_units.not_canceled.reject(&:original_return_item).size)
+    -line_item.price
```

Comment:
Will this work for line items which have promotion adjustmets?

Reply:
Hm hm. Yes and no. Promotion adjustments are `finalized`, and that means they won't get recalculated. The're also no method to find out the line item total before promotions, as cancelations are not part of the line item amount stuff (yet?).

IMO promotions should actually recalculate their amounts when cancelations happen.

Reply:
> IMO promotions should actually recalculate their amounts when cancelations happen.

Yeah I guess there are two separate approaches here, if promotion adjustments are not recalculated, then we need the cancellation adjustment to be for the discounted amount.
If we say they can be recalculated, then we need them to be calculated after the cancellation adjustment is applied on the full price of the item. That will lead to the promo adjustment amount being reduced to only discount non-cancelled quantities from the line.